source/dub/generators/build.d

```diff
@@ -408,6 +445,66 @@ class BuildGenerator : ProjectGenerator {
 		return true;
 	}
 
+	private bool areAllObjectFilesUpToDate(NativePath target_path, BuildSettings buildsettings, GeneratorSettings settings, in Package main_pack, in Package[] packages, in NativePath[] additional_dep_files)
+	{
+		import std.datetime : Clock;
+
+		auto objPath = NativePath(buildsettings.targetPath) ~ NativePath("obj");
+		if (!existsFile(objPath))
+			return false;
+
+		string objectExtension;
+		if (settings.compiler.name == "ldc") {
+			if (buildsettings.dflags.canFind("-output-ll", "--output-ll")) {
+				objectExtension = ".ll";
+			} else if (buildsettings.dflags.canFind("-output-s", "--output-s")) {
+				objectExtension = ".s";
+			} else {
```

Comment:
The logic is unfortunately a bit more complex than that - there's `-output-bc` too, and you can freely combine all 4 switches (`-output-{o,bc,ll,s}`) to generate multiple files per object at once.

Reply:
Also, all of the LDC/DMD specific logic needs to be abstracted away in the `Compiler` interface. Otherwise this will get messy real quick, especially with all other generators potentially needing to duplicate the logic.